cmd/gator/verify/verify.go

```diff
@@ -47,8 +47,8 @@ func init() {
 		`print extended test output`)
 	Cmd.Flags().BoolVarP(&includeTrace, "trace", "t", false,
 		`include a trace for the underlying constraint framework evaluation`)
-	Cmd.Flags().BoolVarP(&flagEnableK8sCel, "experimental-enable-k8s-native-validation", "", false,
-		`PROTOTYPE (not stable): enable the validating admission policy driver`)
+	Cmd.Flags().BoolVarP(&flagEnableK8sCel, "experimental-enable-k8s-native-validation", "", true,
```

Comment:
we can't take any action now, but we should not call flags `experimental` again

Reply:
Should we not? "experimental" sends the message "this may break/go away, do not rely on it", I see no undesireable user impact here. If anything, this means users who were using an experiment are put on notice that its behavior is changing and they should double-check that wont be a problem for them.

Are there impacts other than backwards compatibility? Because, again, backwards compatibility is explicitly not a concern for experimental/alpha things.

Reply:
Also, given this is no longer experimental, we should change the flag name.

Reply:
Changing or removing flags will cause the pod to crash during in place upgrade. That's not a good experience even if the feature is in alpha. A feature flag can be used in the future even after the feature is stable to allow users to opt-out. Therefore, starting a flag with "experiment" is odd if it can be used in the future after the feature is stable.

Reply:
My point is that the flag cannot be used in the future, and that is fine. That is why we used "experimental", we should feel free to change the flag name.

Reply:
Concerns about UX due to breaking changes are specifically excluded for experimental/alpha features. This is because we need room to discover how the feature is used and adjust to feedback and our own observations.

Reply:
@maxsmythe major concern here would be the helm chart. if a user installed through the helm chart, that flag will be set regardless if they opted in or not. when they upgrade through the chart, it will break their installation. we should not break users if they didn't opt in for an experimental feature.

i think we should not add any alpha level features in helm to avoid this issue. obviously, this will make it harder for users to try out new alpha features.

Reply:
> i think we should not add any alpha level features in helm to avoid this issue in the future. obviously, this will make it harder for users to try out new alpha features.

This makes sense to me.

> when they upgrade through the chart, it will break their installation if we remove the flag

TBH if Helm has no way to change flags during upgrade, that points to a pretty large feature gap given running binaries is the majority of the intent of the project. Does Helm really not have a way to change the set of flags provided to a binary? Doesn't "upgrading through the chart" also involve upgrading the chart itself?

Reply:
One option can be to leave the old flag but render it useless. This would avoid crashing. The flag should still go away eventually.

However, I would still like to understand how/why a Helm chart cannot handle a binary changing its flags.

Reply:
> Concerns about UX due to breaking changes are specifically excluded for experimental/alpha features.

This is only true if we plan to never allow users to disable a new feature that is in development. If a new feature can be disabled AFTER it is stable, then it should NOT change or have "experimental" in the flag.

Reply:
I refer you to the K8s API definitions for alpha/beta. They have a policy for flags that say alpha flags can be removed immediately:

https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli

I don't think it's reasonable to hold ourselves to a higher standard than K8s itself.

I also return to the meaning of "experimental"/"alpha" which is:

"The developers are still working on this feature. They make no claim to have any idea as to its final behavior or how it will be configured. Breaking changes are expected"

This isn't about whether we want to continue with a feature gate or not, it's that, at the time of implementation, we were not sure whether/how the feature should work or be gated and wanted to warn the user so that they knew breaking changes are expected.

Whether we want to include "experimental" in a flag name in the future or not... I don't really care much either way. So long as users are put on notice that the API/flag is unstable, the core goal is accomplished.

As to the core question of whether its okay to rename a flag that is in alpha, the answer simply has to be yes. For any reason. We could decide a different name more fully communicates the flag's significance, the underlying behavior the flag controls could change, we could simply not like the name.

Beta is when backwards compatibility becomes a concern, and that is why we want to be careful before graduating a feature to beta, and that is why the flag name needs to change now, because we will not be able to change it in the future -- because we will be held to the requirements of being backwards compatible.

Reply:
I agree if we need to rename a flag before beta it makes sense. i just want to avoid having to rename when we go to beta because we started with `experimental`.

I think both of our concerns can be addressed with:
- return a warning for experimental and deprecating flags
- try not to name flags with experimental to avoid having to change it later

If everyone agrees, I suggest renaming this flag to `nable-k8s-native-validation` since we are moving to beta.

Reply:
Makes sense to me.